File: src/259.3sum-smaller.cs
public class Solution {
    public int ThreeSumSmaller(int[] nums, int target)
    {
        int m = nums.Length;
        List<int> list = nums.ToList();
        list.Sort();
        int result = 0;
        for (int ptr1 = 0; ptr1 < m - 2; ++ptr1)
        {
            int ptr2 = ptr1 + 1;
            int ptr3 = m - 1;
            int targetN = target - list[ptr1];
            int sum = 0;
            while (ptr2 < ptr3)
            {
                sum = list[ptr2] + list[ptr3];
                if (sum < targetN)
                {
                    result += ptr3 - ptr2;
                    ptr2++;
                }
                else
                    ptr3--;
            }
        }
        return result;
    }
}

File: dump-cn/259.3sum-smaller.cpp
class Solution {
public:
	int threeSumSmaller(vector<int>& nums, int target) {
		int result = 0;
		sort(nums.begin(), nums.end());
		for (int i = 0; i < nums.size(); ++i) {
			int t = target - nums[i];
			int ptr = nums.size() - 1;
			for (int j = i + 1; j < nums.size(); ++j) {
				int t2 = t - nums[j];
				while (ptr > j && nums[ptr] >= t2)
					--ptr;
				if (ptr > j)
					result += ptr - j;
			}
		}
		return result;
	}
};